openbas-api/src/main/java/io/openbas/rest/settings/PlatformSettingsApi.java

```diff
@@ -157,6 +159,39 @@ private Setting resolveFromMap(Map<String, Setting> dbSettings, String themeKey,
         return new Setting(themeKey, value);
     }
 
+
+    /**
+     * Return the version of Rabbit MQ we're using
+     *
+     * @return the rabbit MQ version
+     */
+    private String getRabbitMQVersion() {
+        // If we already have the version, we don't need to get it again
+        if (rabbitMQVersion == null) {
+            RestTemplate restTemplate = new RestTemplate();
+            // Init the rabbit MQ management api overview url
+            String uri = this.openBASConfig.isRabbitmqSsl() ? "https://" : "http://"
+                    + this.openBASConfig.getRabbitmqHostname() + ":" + this.openBASConfig.getRabbitmqManagementPort()
+                    + "/api/overview";
+
+            // Init the headers
+            HttpHeaders headers = new HttpHeaders();
+            headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+            headers.setBasicAuth(this.openBASConfig.getRabbitmqUser(), this.openBASConfig.getRabbitmqPass());
+            HttpEntity<String> entity = new HttpEntity<>("parameters", headers);
+
+            // Make the call
+            ResponseEntity<?> result =
+                    restTemplate.exchange(uri, HttpMethod.GET, entity, String.class);
```

Comment:
Can we move this logic to a dedicated service or client class for Rabbit ? 
What do you think ?

Reply:
I was trying to reduce as much as possible the footprint of this code as this is not very elegant but that would make sense to have a dedicated class (helper ?) for this

Reply:
I vote for it, the code can be more readable/separated.